sentry-android-core/src/test/java/io/sentry/android/core/ConnectivityCheckerTest.kt

```diff
@@ -1,31 +1,161 @@
+    }
+
+    @Test
+    fun `When ConnectivityManager is not available, return null for isConnected`() {
+        assertNull(ConnectivityChecker.isConnected(mock(), mock()))
+    }
+
+    @Test
+    fun `When ConnectivityManager is not available, return null for getConnectionType`() {
+        val buildInfo = mock<IBuildInfoProvider>()
+        whenever(buildInfo.sdkInfoVersion).thenReturn(Build.VERSION_CODES.M)
+        assertNull(ConnectivityChecker.getConnectionType(mock(), mock(), buildInfo))
+    }
+
+    @Test
+    fun `When sdkInfoVersion is not min Marshmallow, return null for getConnectionType`() {
+        val buildInfo = mock<IBuildInfoProvider>()
+        whenever(buildInfo.sdkInfoVersion).thenReturn(0)
+        assertNull(ConnectivityChecker.getConnectionType(mock(), mock(), buildInfo))
+    }
+
+    @Test
+    fun `When there's no permission, return null for getConnectionType`() {
+        val contextMock = mock<Context>()
+        val cm = mock<ConnectivityManager>()
+        whenever(contextMock.getSystemService(any())).thenReturn(cm)
+        whenever(contextMock.checkPermission(any(), any(), any())).thenReturn(PERMISSION_DENIED)
+        val buildInfo = mock<IBuildInfoProvider>()
+        whenever(buildInfo.sdkInfoVersion).thenReturn(Build.VERSION_CODES.M)
+        assertNull(ConnectivityChecker.getConnectionType(contextMock, mock(), buildInfo))
+    }
 
-    @BeforeTest
-    fun `set up`() {
-        context = ApplicationProvider.getApplicationContext()
+    @Test
+    fun `When network is not active, return null for getConnectionType`() {
+        val contextMock = mock<Context>()
+        val cm = mock<ConnectivityManager>()
+        whenever(contextMock.getSystemService(any())).thenReturn(cm)
+        val buildInfo = mock<IBuildInfoProvider>()
+        whenever(buildInfo.sdkInfoVersion).thenReturn(Build.VERSION_CODES.M)
+        assertNull(ConnectivityChecker.getConnectionType(contextMock, mock(), buildInfo))
+    }
+
+    @Test
+    fun `When network capabilities are not available, return null for getConnectionType`() {
+        val contextMock = mock<Context>()
+        val cm = mock<ConnectivityManager>()
+        whenever(contextMock.getSystemService(any())).thenReturn(cm)
+        val net = mock<Network>()
+        whenever(cm.activeNetwork).thenReturn(net)
+        val buildInfo = mock<IBuildInfoProvider>()
+        whenever(buildInfo.sdkInfoVersion).thenReturn(Build.VERSION_CODES.M)
+        assertNull(ConnectivityChecker.getConnectionType(contextMock, mock(), buildInfo))
+    }
+
+    @Test
+    fun `When network capabilities has TRANSPORT_WIFI, return wifi`() {
+        val contextMock = mock<Context>()
+        val cm = mock<ConnectivityManager>()
+        whenever(contextMock.getSystemService(any())).thenReturn(cm)
+        val net = mock<Network>()
+        whenever(cm.activeNetwork).thenReturn(net)
+        val nc = mock<NetworkCapabilities>()
+        whenever(cm.getNetworkCapabilities(any())).thenReturn(nc)
```

Comment:
`m`: Can we please make use of a `@Before` method for all tests here. The tests are hard to read IMO. A little bit of a splitting up into either arrange, act, assert or given, when, then would be nice.

Reply:
splitting up makes sense, fwiw `act/assert` here are merged as we test a single method and a single assert

Reply:
as stated before, `Before` won't apply when we don't use all the mocks in all the tests, we'd be paying for something not used

Reply:
Still we have lots of duplicated code here and the tests are hard to read IMO. I opened up a PR on your forked repo to show you the difference.